### PR TITLE
Add all inline elements to avoid wrong space breaking

### LINKED ISSUE #75: xexpr->string screws up whitespace
Consider:

```racket
#lang racket

(require (only-in markdown [xexpr->string f1]))
(require (only-in xml [xexpr->string f2]))

(define (xexprs->string f xs)
  (string-join (map f xs) "\n"))

(define data '((p () "Hello" (label) ".")))

(xexprs->string f1 data)
(xexprs->string f2 data)
```

Here's the result:

```
"\n<p>Hello\n <label></label>.</p>"
"<p>Hello<label></label>.</p>"
```

When rendered, `xexpr->string` from this package will produce an extraneous whitespace which is incorrect. Since Frog uses this function, HTML rendering in Frog is also wrong.

### PR DESCRIPTION
**Pull request by sorawee**
Fix #75

### CODE REVIEW

**Comment by sorawee** [comment]
Note: the failure in travis-ci is due to nondeterminism of the setup. It's not related to my PR. You can request for a rebuild (I don't have a permission to do so), and it probably will now pass.

**Comment by greghendershott** [comment]
I've been swamped and only recently was able to dig in and think about this.

- I like the idea of using the updated HTML5 inline elements.

- I'm unsure about changing the behavior so that there isn't a leading newline, anymore.

Many years ago when I was new to Racket, I found it strange that this writes a leading newline:

```racket
(display-xml/content
 (document-element
  (read-xml (open-input-string
             "<html></html>"))))
```
=>

```

<html>
</html>
```

And then in the Racket web-server I saw it wrote a doctype line, followed by `display-xml`. The leading newline seemed to be an intentional feature.

So when I did markdown for use in frog, I emulated this in `display-xexpr`.

That was what was nagging at me, about this PR. I knew there was _something_, but I couldn't find time to focus and recall that.

---

Fast forward some years, and frog doesn't even use `display-xexpr` anymore to write the page. It switched to using webserver/templates. So the change would be N/A for Frog.

However I wonder if it would break anything else. e.g. Pollen uses this package.

---

TL;DR I think the PR is good, except we should keep the leading newline.  If that really bugs you, then we should add an optional keyword argument for it, which defaults to true for backward compatibility.

**Comment by greghendershott** [comment]
@sorawee If you're reading this in email, the previous comment is better read here on GitHub -- I made a few edits to clarify. I don't think GitHub re-sends edits as emails.
